hub/src/server/drivers/diskDriver.ts

```diff
@@ -274,8 +283,15 @@ class DiskDriver implements DriverModel {
     const contentTypeJsonStr = await fs.readFile(contentTypeFilePath, 'utf8')
     const contentType = JSON.parse(contentTypeJsonStr)['content-type']
     const lastModified = dateToUnixTimeSeconds(stat.mtime)
+
+    // create "weak" tag from last-modified time and file size
+    const mtime = stat.mtime.getTime().toString(16)
```

Comment:
I think this should just store a UUID as the etag within the corresponding json file metadata. We already eat all the costs from maintaining metadata state due to `content-type`. 
```js
// TODO: replace `contentTypeJsonStr` with something less specific 
const metadata = JSON.prase(contentTypeJsonStr)
const contentType = metadata['content-type']
const etag = metadata['etag']
```